BSP/Simulator/Hardware/simulate.py

```diff
@@ -15,6 +16,35 @@ def update_buttons():
             button.config(relief=tk.RAISED)
     window.after(1, update_buttons)
 
+def update_timer():
+	""" Updates Timer.csv every 1 ms """
+
+	currentValue = 0 # If exception is caught, skip writing by initializing to 0
+	simulatorFile = "Timer.csv"
+
+	try:
+		with open(simulatorFile, 'r') as csvfile:
+			reader = csv.reader(csvfile, delimiter=',')
+			
+			try:
+				for row in reader: # Should only be a single row
+					currentValue = int(row[0])
+					reloadValue = int(row[1]) # Save so we can rewrite below
+
+			except IndexError: # If the csv is blank. Not sure what to do in this case so I just print for debugging.
+				print("Blank CSV")
+
+
+		if currentValue > 0: # Systick stops at 0
+			with open(simulatorFile, 'w') as csvfile:
+				writer = csv.writer(csvfile, delimiter=',')
+
+				writer.writerow([currentValue - 1, reloadValue])
+
+	except FileNotFoundError:
+		print("File " + simulatorFile + " not found")
+	
+	window.after(1, update_timer)
```

Comment:
Most of this functionality should be implemented in a separate Timer.py file. Take a look at the Switches.py and Pedals.py files to see the formatting for that. For the exceptions, we want to create the file if they are not already created (again, see the other modules for examples). In terms of the Blank CSV situation, if there is a blank file (meaning nothing has been initialized), just set the reloadValue and currentValue to some maximum value (like 0xFFFFFF)